tests/helpers/utils.py

```diff
@@ -111,3 +115,18 @@ def inner_f(queue, **kwargs):
         assert result == 1, 'expected 1, but returned %s' % result
 
     return wrapper
+
+
+@contextmanager
+def no_warning_call(warning_type, match: Optional[str] = None):
+    with pytest.warns(None) as record:
+        yield
+
+        try:
+            w = record.pop(warning_type)
+            if not ((match and match in str(w.message)) or w):
```

Comment:
I think these parentheses are redundant. Also that there is `.text` but haven't run it

```suggestion
            if not (match and match in w.text or w):
```